graphql/src/execution/execution.rs

```diff
@@ -199,11 +201,20 @@ lazy_static! {
         .expect("Invalid value for GRAPH_QUERY_CACHE_STALE_PERIOD environment variable")
     };
 
+    /// In how many shards (mutexes) the query block cache is split.
+    static ref QUERY_BLOCK_CACHE_SHARDS: u8 = {
+        std::env::var("GRAPH_QUERY_BLOCK_CACHE_SHARDS")
+        .unwrap_or("100".to_string())
+        .parse::<u8>()
+        .expect("Invalid value for GRAPH_QUERY_BLOCK_CACHE_SHARDS environment variable, max is 255")
+    };
+
 
     // Cache query results for recent blocks by network.
     // The `VecDeque` works as a ring buffer with a capacity of `QUERY_CACHE_BLOCKS`.
-    static ref QUERY_BLOCK_CACHE: TimedMutex<QueryBlockCache> =
-                                    TimedMutex::new(QueryBlockCache(vec![]), "query_block_cache");
+    static ref QUERY_BLOCK_CACHE: Vec<TimedMutex<QueryBlockCache>> =
+                                    std::iter::repeat_with(|| TimedMutex::new(QueryBlockCache(vec![]), "query_block_cache"))
```

Comment:
What do you think of including the number of the shard in the identifier for the mutex, something like `query_block_cache_42`?